accounts/usbwallet/wallet.go

```diff
@@ -338,14 +338,15 @@ func (w *wallet) selfDerive() {
 		}
 		// Device lock obtained, derive the next batch of accounts
 		var (
-			accs  []accounts.Account
-			paths []accounts.DerivationPath
-
-			nextPaths = append([]accounts.DerivationPath{}, w.deriveNextPaths...)
-			nextAddrs = append([]common.Address{}, w.deriveNextAddrs...)
-
+			accs    []accounts.Account
+			paths   []accounts.DerivationPath
 			context = context.Background()
+			nextAddrs = append([]common.Address{}, w.deriveNextAddrs...)
 		)
+		nextPaths := make([]accounts.DerivationPath, len(w.deriveNextPaths))
+		for i, path := range w.deriveNextPaths {
+			nextPaths[i] = append(accounts.DerivationPath(nil), path...)
+		}
```

Comment:
I like the idea of not updating slices that are aliased to w.deriveNextPaths while holding stateLock.RLock. From reading the current code, it seems like SelfDerive is the only other user of deriveNextPaths and it always runs under stateLock.Lock. I could be wrong, but I’m not seeing an actual concurrent read/write.

Did you hit a -race report in practice, or is this more of a defensive change to avoid future issues?